.github/workflows/production.yml

```diff
@@ -0,0 +1,50 @@
+name: CI_CD
+
+on:
+  push:
+    branches: [main]
+  workflow_dispatch:
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [16.x]
+    steps:
+    - name: Run a one-line script
+      run: echo Hello, world!
+
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v2
+      with:
+        node-version: ${{ matrix.node-version }}
+        cache: 'yarn'
+
+    - name: Install dependencies
+      run: yarn install
+
+    - name: Test linting
+      run: yarn run lint
+
+    - name: Run Unit Tests
+      run: yarn run test
+
+    - name: Run Coverage
+      run: npm run test:coverage
+
+    - name: generate Coverage report
+      run: npm run coverage:ci
+
+    - name: Check coverage threshold.
```

Comment:
checks for coverage. Some default files are not giving more than 30% coverage so I put 30% for now.